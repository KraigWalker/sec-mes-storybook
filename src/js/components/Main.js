import React from 'react';
import { connect } from 'react-redux';


class Main extends React.PureComponent {
    render() {
        return (
            <div>
                <h1>
                Secure Message Web App
                </h1>
            <div>
                { this.props.children }
            </div>
        </div>);
    }
}

const mapState = (state) => {
    return state
} 

export default connect(mapState)(Main);