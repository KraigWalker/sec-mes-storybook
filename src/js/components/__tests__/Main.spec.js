import { Main } from '../Main';
import { shallow } from 'enzyme';

describe('Main snapshot', () => {
  let props = {
    children: 'cjejd',
    location: {
      pathname: '',
    },
  };
  let component = shallow(<Main {...props} />);
  it('should match to snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
