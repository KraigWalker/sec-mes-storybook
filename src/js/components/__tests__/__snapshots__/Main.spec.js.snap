// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main snapshot should match to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    location={
      Object {
        "pathname": "",
      }
    }
  >
    cjejd
  </Main>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        undefined,
        "cjejd",
      ],
    },
    "ref": null,
    "rendered": Array [
      undefined,
      "cjejd",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          undefined,
          "cjejd",
        ],
      },
      "ref": null,
      "rendered": Array [
        undefined,
        "cjejd",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "prevContext": true,
          },
          "getChildContext": Object {
            "calledByRenderer": true,
          },
        },
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
